src/OrchardCore.Modules/OrchardCore.Media/Assets/js/MediaApp/app.js

```diff
@@ -80,6 +80,7 @@ function initializeMediaApplication(displayMediaApplication, mediaApplicationUrl
                         
                         media.mediaPath = newPath;
                         media.name = newName;
+                        media.url = media.url.substr(0, media.url.lastIndexOf(oldPath)).concat(newPath);
```

Comment:
Thanks for having used the gulp pipeline!

Here i would suggest

    media.url = media.url.substr(0, media.url.lastIndexOf('/') + 1) + encodeURI(newName);

So that if the filename contains a space it would be replaced by `%20`. But anyway it is not done on server side when we first get the urls, and the browser does it for us, maybe not all browsers but we would have to also fix it on server side.

So for simplicity i suggest

    media.url = media.url.substr(0, media.url.lastIndexOf('/') + 1) + newName;